apps/examples/e2e/tests/test-camera.spec.ts

```diff
@@ -8,12 +22,124 @@ export function sleep(ms: number) {
 test.describe('camera', () => {
 	test.beforeEach(setup)
 
-	test.fixme('panning', () => {
-		// todo
+	test('panning', async ({ isMobile, page }) => {
+		test.skip(!isMobile)
+		client = await page.context().newCDPSession(page)
+
+		expect(await page.evaluate(() => editor.inputs.currentPagePoint)).toEqual({
+			x: 50,
+			y: 50,
+			z: 0,
+		})
+
+		await dispatchTouch(client, 'touchStart', [
+			{ x: 100, y: 100 },
+			{ x: 200, y: 200 },
+		])
+		let finalTouch: { x: number; y: number }[] = [
+			{ x: 100, y: 100 },
+			{ x: 200, y: 200 },
+		]
+		for (let i = 1; i < 100; i++) {
+			await dispatchTouch(client, 'touchMove', [
+				{ x: 100 + i * 10, y: 100 + i * 10 },
+				{ x: 200 + i * 10, y: 200 + i * 10 },
+			])
+			finalTouch = [
+				{ x: 100 + i * 10, y: 100 + i * 10 },
+				{ x: 200 + i * 10, y: 200 + i * 10 },
+			]
+			await sleep(10)
+		}
+		await dispatchTouch(client, 'touchEnd', finalTouch)
+
+		await sleep(10)
+		// not sure how the touch coordinates translate to the editor coordinates
+		expect(
+			await page.evaluate(() => [
+				editor.inputs.currentPagePoint.x,
+				editor.inputs.currentPagePoint.y,
+			])
+		).toStrictEqual([160, 160])
 	})
 
-	test.fixme('pinching', () => {
-		// todo
+	test('pinching on trackpad', async ({ page, isMobile }) => {
+		// pinching on trackpad is the same event as ctrl+scrollwheel
+		test.skip(isMobile)
+		await page.evaluate(() => editor.updateInstanceState({ isGridMode: true }))
+		const { mouse, keyboard } = page
+		expect(await page.evaluate(() => editor.getZoomLevel())).toBe(1)
+		// zooming in
+		await keyboard.down('Control')
+		// scrolling 15px on the mouse wheel is a bit less that 0.1 zoom level at default zoom speed
+		await mouse.wheel(0, 15) // 0.9
+		await mouse.wheel(0, 15) // 0.81
+		await mouse.wheel(0, 15) // 0.72
+		await keyboard.up('Control')
```

Comment:
This is odd, if I await a single mouse.wheel event, even if I set the delta y to 1000000, it only zooms to 0.9. It maxes out at 15 pixels, any more and there is no further change.